app/javascript/readingList/collectionForm.jsx

```diff
@@ -0,0 +1,87 @@
+import { h, Component } from 'preact';
+import PropTypes from 'prop-types';
+import linkState from 'linkstate';
+import Tags from '../shared/components/tags';
+
+export class CollectionForm extends Component {
+  constructor({ tagList }) {
+    super({ tagList });
+    this.state = {
+      // eslint-disable-next-line react/no-unused-state
+      title: '',
+      tagList,
+      // auth: false
+    };
+  }
+
+  handleChange = e => {
+    this.setState({ [e.target.name]: e.target.value });
+  };
+
+  submitCollection = (e, title, tags) => {
+    // check that tag length is > 1, title req handled on input
+    // need to modify route also and ensure error handling is included.
+
+    // if successful route to readinglist               href="/readinglist"
+    e.preventDefault();
+    const tagArr = tags.split(',');
+    if (title.length && tags.length) {
+      fetch('/api/reading_collections', {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'X-CSRF-Token': window.csrfToken,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
```

Comment:
Nice work on getting this form done! We talked "in-person" about cleaning up how the tags are coming in on a future PR.